scipy/special/tests/test_basic.py

```diff
@@ -2088,55 +2089,79 @@ def test_expm1more(self):
         assert_array_almost_equal(ex1,exrl1,8)
 
 
+def assert_really_equal(x, y, rtol=None):
```

Comment:
Are we will willing to use `xp_assert_*` also for regular functions that don't use anything about the array API?

Also, would people be willing to enhance the strict check to distinguish nans vs. complex nans? I'm not introducing this in this PR, but I need something along the lines of https://github.com/scipy/scipy/pull/19988/commits/ed245ece051f4dc21d1cb406f446131b49c2f9f3 (see #19988). If I should incorporate that into `xp_assert_{equal,close}`, that's OK for me as well.

Reply:
> would people be willing to enhance the strict check to distinguish nans vs. complex nans?

cc @mdhaber 

> Are we will willing to use xp_assert_* also for regular functions that don't use anything about the array API?

From my perspective it makes sense since ultimately they will be replaced with xp-assertions anyway. The only superfluous check for now can be turned off with `check_namespace=False`.

Reply:
> enhance the strict check to distinguish nans vs. complex nans

Do you mean:
- distinguishing between 
  - a complex number `x` for which `np.isnan(x)` is True

    and
  - a complex number `x` for which `np.isnan(x.real) and np.isnan(x.imag)` is True
 
  or 
- distinguishing between
  - a number `x` for which `np.isnan(x)` is True

    and

  - a number `x` for which `np.isnan(x) and np.issubdtype(x.dtype, np.complexfloating)` is True

?

Reply:
The type equality is already included in the strict mode, but I want to distinguish `NaN` from `NaN + 0j` from `NaN + NaN*j`, so this:

> a complex number `x` for which `np.isnan(x.real) and np.isnan(x.imag)` is True

Reply:
OK. Yeah, that sounds OK.